sbt-metals/src/main/scala/scala/meta/sbt/MetalsPlugin.scala

```diff
@@ -49,8 +31,19 @@ package scala.meta.sbt {
         )
       }
     }
+
     import autoImport._
 
+    val metalsBuildInfo = taskKey[Map[String, String]](
+      "List of key/value pairs for build information such as classpath/sourceDirectories"
+    )
+    val metalsWriteBuildInfo = taskKey[Unit](
+      "Write build information to .metals/buildinfo/"
+    )
+
+    val semanticdbScalac =
+      "org.scalameta" % "semanticdb-scalac" % Metals.semanticdbVersion cross CrossVersion.full
+
```

Comment:
I suggest moving these values to the `Metals` object

Reply:
Is `Metals` in scope from build.sbt? I would like to call `Metals.semanticdbScalac`? I kept these here so that users only need to be aware of `MetalsPlugin`, which is already a familiar concept for users

Reply:
Yes, `scala.meta.sbt.Metals` is in scope in `build.sbt` and it contains some other values that could be useful if someone want to customize or inspect things (like `semanticdbVersion`?), so I'd keep them together. If you move these values to `Metals`, then users don't need to know anything about `MetalsPlugin` (they are not familiar with it, because it's a global autoplugin which they don't need to enable manually).

Reply:
About global/local: if users refer to `Metals.semanticdbVersion` in their `build.sbt` then they will have to install sbt-metals locally, or the build won't compile for a co-worker who's not using metals.

Reply:
This effectively means we'll be requiring users to install `sbt-metals` per-project, which sucks because it makes updating the "IDE" a project-wide operation, rather than a dev operation.

Can we make this dynamic somehow? I.e. only enable `semanticdbScalac` if the plugin is globally installed or bail otherwise?

Reply:
> About global/local

Right. Anyway, I would put all reusable values in an external object to keep the plugin definition minimal. So if user wants to use anything, they can refer to `scala.meta.sbt.Metals`, or if you don't like the package prefix, add an alias to the `autoImport`: `val Metals = scala.meta.sbt.Metals`.

> This effectively means we'll be requiring users to install sbt-metals per-project

I think the only problem here is in case when you need to setup Metals on custom configurations in your project. If you only use `Compile` and `Test`, you can keep sbt-metals global and only call `metalsSetup` on the projects you want. Or am I missing some point about `semanticdbScalac`?

Reply:
We can't avoid the problem that any IDE tooling that depends on sbt will need to be defined in the build. A user can `addCompilerPlugin` in some `ignore.sbt` file if they don't want to pollute the build definition. My hope is the benefits from semanticdb-scalac will soon be so many that teams will decide to enable it by default to enjoy

* Scalafix
* Metals

Reply:
How about this? https://gitter.im/sbt/sbt?at=5acb65ff92f5d62057ac694d

Reply:
We can make a

Reply:
I'm sorry, @gabro, could you explain what problem are you trying to solve?

Reply:
I don't want users to add `sbt-metals` to their project's build.

I would love for them to have it as a global plugin only, so that it's easier to update.

Reply:
OK. This seems to be the initial plan. In what cases users would _need_ to add this plugin per-project? 

I see only the case if they have some custom (sbt-)configuration, for which they want to generate metals buildinfo. Yet, I think it's enough to add a comment to the docs: 

> if you have a custom configuration `Foo` in project `buh`, in addition to `metalsSetup` also run `bug/Foo/metalsWriteBuildInfo` in sbt

i.e. user can still do this in the sbt shell without the need to add anything to their build. And if they want, they can use `ignore.sbt` workaround.

Reply:
I've moved the settings to the `Metals` object. I personally prefer to keep everything in `MetalsPlugin` since it's consistent with BuildInfoPlugin, ScalafixPlugin, etc.

Can we move this discussion about sbt-metals and project-specific installation to a dedicated ticket? This PR was primarily about avoiding an import conflict with sbt-scalafix.

Reply:
>  In what cases users would need to add this plugin per-project

Whenever they add `Metals.semanticdbScala` to a `build.sbt`.

If you don't include the plugin per-project, then you're potentially breaking your co-worker's (and the CI's) builds

Reply:
Yes, let's merge this move the discussion somewhere else.

Reply:
@laughedelic It doesn't seem like `Metals` is autoimported into scope

```scala
/Users/ollie/dev/metals/build.sbt:71: error: not found: value Metals
    addCompilerPlugin(Metals.semanticdbScalac),
                      ^
```

I propose we stick to having everything in `MetalsPlugin` as that's automatically in scope.

Reply:
@olafurpg see above: https://github.com/scalameta/metals/pull/272#discussion_r180085310, you need to refer to it with the package prefix or define an auto-imported alias. But anyway, I don't have a strong opinion, so feel free to change it back if you prefer.